core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -831,6 +831,7 @@ class ReplicaManager(val config: KafkaConfig,
             error match {
               case Errors.INVALID_TXN_STATE => Some(error.exception("Partition was not added to the transaction"))
               case Errors.CONCURRENT_TRANSACTIONS |
+                   Errors.NETWORK_EXCEPTION |
```

Comment:
Should we add a comment why we're translating to NotEnoughReplicasException error and not something else?  I think there is a bit of thinking that went into this logic and the code doesn't reflect it.

Reply:
Good idea, I've added a comment.